File: src/is-hex-digit.js
/**
 * @fileoverview
 * Helper function to check if a character is a hexadecimal digit.
 */

/**
 * Checks if a character is a hex digit (0-9, a-f, A-F).
 * @param {string} ch - Single character
 * @returns {boolean}
 */
export function isHexDigit(ch) {
    const code = ch.charCodeAt(0);
    return (code >= 48 && code <= 57) // 0-9
        || (code >= 65 && code <= 70) // A-F
        || (code >= 97 && code <= 102); // a-f
}

File: wasm/jsonc_lexer/src/is_hex_digit.rs
//! Helper function to check if a byte is a hexadecimal digit.

/// Checks if a byte is a hex digit (0-9, a-f, A-F).
pub fn is_hex_digit(ch: u8) -> bool {
    (ch >= 48 && ch <= 57)  // 0-9
        || (ch >= 65 && ch <= 70)  // A-F
        || (ch >= 97 && ch <= 102) // a-f
}
